src/plugins/kibana_react/public/field_button/field_button.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import classNames from 'classnames';
+import React, { ReactNode } from 'react';
+import './field_button.scss';
+
+export interface FieldButtonProps {
+  isOpen?: boolean;
+  fieldIcon?: ReactNode;
+  fieldName?: ReactNode;
+  fieldInfoIcon?: ReactNode;
+  fieldAction?: ReactNode;
+  isDraggable?: boolean;
+  size?: ButtonSize;
+  className?: string;
+  onClick?: () => void;
+}
+
+/**
+ * Wraps Object.keys with proper typescript definition of the resulting array
+ */
+function keysOf<T, K extends keyof T>(obj: T): K[] {
+  return Object.keys(obj) as K[];
+}
+
+export type ButtonSize = 's' | 'm';
+
+const sizeToClassNameMap: { [size in ButtonSize]: string | null } = {
+  s: 'kbnFieldButton--small',
+  m: null,
+};
+
+export const SIZES = keysOf(sizeToClassNameMap);
+
+export function FieldButton({
+  size = 'm',
+  isOpen,
+  fieldIcon,
+  fieldName,
+  fieldInfoIcon,
+  fieldAction,
+  className,
+  isDraggable = false,
+  onClick,
+  ...rest
+}: FieldButtonProps) {
+  const classes = classNames(
+    'kbnFieldButton',
+    size ? sizeToClassNameMap[size] : null,
+    { 'kbnFieldButton-isOpen': isOpen },
+    { 'kbnFieldButton--isDraggable': isDraggable },
+    className
+  );
+
+  return (
+    <div className={classes} {...rest}>
+      <button onClick={onClick} className="kbn-resetFocusState kbnFieldButton__content">
+        <div className="kbnFieldButton__fieldIcon">{fieldIcon}</div>
+        <div className="kbnFieldButton__name">{fieldName}</div>
+      </button>
+      <div className="kbnFieldButton__append">
+        {fieldAction && <div className="kbnFieldButton__fieldAction">{fieldAction}</div>}
+        {fieldInfoIcon && <div className="kbnFieldButton__infoIcon">{fieldInfoIcon}</div>}
+      </div>
```

Comment:
But we can't nest interactive content within each other... Can we instead add a click handler to the icon which does the same as the button?

Reply:
This is where technicality comes into question. Technically, the icon is **not** interactive, it has no click behavior only hover behavior to show extra help text that refers to the whole button's action. What we're achieving with this icon is to limit the hoverable area that will show the tooltip so you can't get an accordion affect of tooltips when mousing over the list.

Reply:
Aren't all of our hoverables also focusable? Otherwise, how else does a keyboard-only user navigate to it? And if it's focusable, we get back to not being able to nest focusable elements.

If it's not focusable, we could show this tooltip on button focus as well and then wouldn't have to worry about this icon not being reachable by keyboard.

Reply:
It doesn't need to be focusable because the tooltip content surrounding the icon is the same as the aria-label for the whole button, so it would actually be redundant. The icon and tooltip are actually more for sighted users.

Again, we don't want the tooltip on the entirety of the button because it would cause the accordion affect of showing many tooltips while scanning of the list with the mouse.

Reply:
Keyboard-only doesn't necesserily include a screen reader though so we still need it available on the screen without requiring a mouse.

So I think this combination should avoid the accordion effect while still meeting the other criteria:
- Button, on focus and not on hover, shows tooltip
- Info icon, on hover and not on focus, shows tooltip

Reply:
I created a PR so "fix" the issue: https://github.com/andreadelrio/kibana/pull/3

To reiterate what I say in that summary:

> This does not completely solve the case for a sighted, but keyboard-using user (without screen reader). But this doesn't hinder us from getting there and I think we need more support from the EuiTooltip component to get us there (ie being able to say `onFocusOnly`).

> This isn't a degraded experience for those users compared to master, so in order to push along the original intent of the PR, I'd opt that we circle back to the sighted, but keyboard-using user experience.